app/models/user.rb

```diff
@@ -318,4 +318,20 @@ def no_other_logins_exist?(login)
   def uploaded_subjects_count
     Subject.unscoped.where(upload_user_id: self.id).count
   end
+
+  def anonymize
+    self.email = "noreply-#{SecureRandom.hex(4)}@zooniverse.org"
+    self.current_sign_in_ip = nil
+    self.last_sign_in_ip = nil
+    self.display_name = self.id
+    self.login = self.id
```

Comment:
Validation and indexes would stop us from saving that. Not too worried about that, unlikely to happen and it's a behind the scenes management function anyways.